molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,993 @@
+                        provider='Libvirt'))
+            else:
+                status_list.append(
+                    Status(
+                        name=instance['name'],
+                        state='undefined',
+                        provider='Libvirt'))
+
+        return status_list
+
+    def _lease_ip(self, interface):
+        """
+        Lease IP address
+
+        :param interface: an interface dict that has at least: {'mac': MAC address, 'network_name': (libvirt) name of network}
+        :return: an IP address
+        """
+        delay = 1
+        max_count = 6
+        net = self._libvirt.networkLookupByName(interface['network_name'])
+        count = 0
+        while not 'ip' not in interface:
+            count = count + 1
+            if count > max_count:
+                return None
+            utilities.print_info(
+                "\t\t Waiting for DHCP for network: {}".format(interface[
+                    'network_name']))
+            time.sleep(delay + count)
+            leases = net.DHCPLeases()
+            if not leases:
+                time.sleep(delay + count)
+                continue
+            for lease in leases:
+                if lease['mac'] == interface['mac']:
+                    utilities.print_success(
+                        "\t\tIP leased for mac {} on network: {}: {}".format(
+                            interface['mac'], interface['network_name'], lease[
+                                'ipaddr']))
+                    interface['ip'] = lease['ipaddr']
+                    return interface['ip']
+
+    def _nics(self, domain):
+        """
+        Retrieve a list of network interfaces
+
+        :param domain: a libvirt domain object
+        :return: list of interfaces
+        """
+        dom = ET.fromstring(domain.XMLDesc())
+        nics = [e for e in dom.findall('./devices/interface[mac]')]
+        return nics
+
+    def _get_ip(self, mac):
+        """
+        Search libvirt's networks' leases for this MAC, return its IP address
+
+        :param mac: a MAC address, formatted as in libvirt's XML: 'AA:BB:CC:DD:EE:FF'
+        :return: an IPv4 address
+        """
+        nets = self._libvirt.listAllNetworks()
+        leases = sum((net.DHCPLeases() for net in nets), [])
+        return next((lease['ipaddr'] for lease in leases
+                     if lease['mac'] == mac), None)
+
+    def _macs(self, domain, network):
+        """
+        Get a Domain's MAC addresses for a particular network.
+
+        :param domain: a libvirt domain object
+        :param network: the name of the network
+        :return: a MAC address that this domain has on the specified network
+        """
+        macs = []
+        nics = self._nics(domain)
+        for nic in nics:
+            mac = nic.find('./mac[@address]').get("address")
+            source_net = nic.find('./source[@network]').get("network")
+            if source_net == network:
+                macs.append(mac)
```

Comment:
Since you're returning the first mac anyway you can simply `return mac` here and completely delete the if/else at the end.